unit_tests/test_exceptions.py

```diff
@@ -62,6 +62,37 @@ def test_hit_w_content_as_str(self):
         self.assertEqual(exception.message, 'Not Found')
         self.assertEqual(list(exception.errors), [])
 
+    def test_hit_w_content_as_unicode(self):
+        from google.cloud.exceptions import NotFound
+        _ERROR_MESSAGE = u'That\u2019s not found.'
+        _EXPECTED = (b'404 That\xe2\x80\x99s not found.',
+                     '404 %s' % (_ERROR_MESSAGE,))
+
+        response = _Response(404)
+        content = u'{"error": {"message": "%s" }}' % (_ERROR_MESSAGE,)
+
+        exception = self._callFUT(response, content)
+        self.assertIn(str(exception), _EXPECTED)
+        self.assertIsInstance(exception, NotFound)
+        self.assertEqual(exception.message, _ERROR_MESSAGE)
+        self.assertEqual(list(exception.errors), [])
+
+    def test_hit_w_content_as_unicode_as_py3(self):
+        import six
+        from unit_tests._testing import _Monkey
+        from google.cloud.exceptions import NotFound
+        _ERROR_MESSAGE = u'That\'s not found.'
```

Comment:
I can't use the non `ascii` characters here because this test skips the `_to_bytes` call in `GoogleCloudError.__str__`.